frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringWorkloadFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import * as fuzzy from 'fuzzysearch';
+import { DeploymentModel, StatefulSetModel, DaemonSetModel } from '@console/internal/models';
+import { Firehose } from '@console/internal/components/utils';
+import { ResourceDropdown } from '@console/shared';
+
+export enum OptionTypes {
+  selectAll = '#SELECT_ALL_WORKLOADS#',
+}
+export const OptionValues = {
+  [OptionTypes.selectAll]: 'All Workloads',
+};
+type MonitoringWorkloadFilterProps = {
+  namespace: string;
+  name: string;
+  onChange: (key: string, type: string) => void;
+};
+export const MonitoringWorkloadFilter: React.FC<MonitoringWorkloadFilterProps> = React.memo(
+  ({ namespace, name, onChange }) => {
+    const selectedTaskRef = React.useRef<any>(null);
+    selectedTaskRef.current = name;
```

Comment:
nit: can we have `string` instead of `any` and assign default in same line 
```suggestion
    const selectedTaskRef = React.useRef<string>(name);
```

Reply:
Updated the type and added the default value.